src/scene/lighting/lights-buffer.js

```diff
@@ -249,7 +250,7 @@ class LightsBuffer {
         // textures
         this._lightsTexture8Id.setValue(this.lightsTexture8);
 
-        if (LightsBuffer.lightTextureFormat === LightsBuffer.FORMAT_FLOAT) {
+        if (LightsBuffer.lightTextureFormat === LightsBuffer.FORMAT_FLOAT && this._lightsTextureFloatId) {
```

Comment:
These are fixes for the `NullGraphicsDevice` crashing when you create another Device after it, because of their shared `LightsBuffer` global state. Creating a `NullGraphicsDevice` should not affect this state as it doesn't use the `LightsBuffer`.

Reply:
related: https://github.com/playcanvas/engine/pull/5935